exec/java-exec/src/main/java/org/apache/drill/exec/work/prepare/PreparedStatementProvider.java

```diff
@@ -138,7 +139,9 @@ public void run() {
         final QueryId limit0QueryId = userWorker.submitWork(wrapper, limit0Query);
 
         final long timeoutMillis =
-            userWorker.getSystemOptions().getOption(CREATE_PREPARE_STATEMENT_TIMEOUT_MILLIS).num_val;
+            userWorker.getDrillbitContext().getConfig().getBoolean(ExecConstants.SEPARATE_WORKSPACE)
```

Comment:
As noted above, this kind of code will lead to countless subtle errors. Suggestion: introduce yet another option manager whose job is to decide which of the two "system" option managers is needed. Option managers are designed to nest, so this should be not too hard.

And, there is still the question: why do we need two *system* option managers? Do users get their own system options now? See questions above...